src/pages/index.js

```diff
@@ -55,41 +137,86 @@ const enableValidation = (config) => {
 
 enableValidation(formValidationOptions);
 
-const cardSection = new Section(
-  {
-    items: initialCards,
-    renderer: (data) => {
-      cardSection.addItems(createCard({ name: data.name, link: data.link }));
-    },
-  },
-  selectors.cardSection
-);
+api
+  .getInitialCards()
+  .then((initialCards) => {
+    cardSection.renderItems(initialCards);
+  })
+  .catch((err) => {
+    console.error(err);
+  });
 
 function handleImageClick(data) {
   cardPreviewImage.open(data);
 }
 
-cardSection.renderItems();
-
-const newcardPopup = new PopupWithForm(selectors.newCardModal, (cardData) => {
-  cardSection.addItems(createCard({ name: cardData.title, link: cardData.url }));
-  newcardPopup.close();
-  formValidators["card-form"].disableButton();
+//update avatar
+const avatarModal = new PopupWithForm("#modal-avatar", (userData) => {
+  const avatarSubmitButton = document.querySelector(
+    selectors.avatarSubmitButton
+  );
+  avatarSubmitButton.textContent = "Saving... ";
+  avatarModal.setLoading(true);
+  api
+    .updateAvatar(userData.url)
+    .then((res) => {
+      userInfoData.setAvatar(res.avatar);
+      avatarModal.close();
+      formValidators["avatar-form"].disableButton();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      avatarModal.setLoading(false);
+      avatarSubmitButton.textContent = "Save";
+    });
 });
 
-newcardPopup.setEventListners();
+avatarModal.setEventListeners();
+
 document
-  .querySelector(selectors.addProfilebutton)
-  .addEventListener("click", () => newcardPopup.open());
+  .querySelector(".profile__button-img")
+  .addEventListener("click", () => avatarModal.open());
+
+const newCardPopup = new PopupWithForm(selectors.newCardModal, (cardData) => {
+  const cardSubmmitButton = document.querySelector(selectors.cardSubmitButton);
```

Comment:
It’s better to find **constant DOM elements** only 1 time at the top of the file so as not to waste resources on searching them again and again when you call a method (function) because searching is a very **hard** operation for the browser engine